nextcord/application_command.py

```diff
@@ -745,12 +745,50 @@ def from_callback(
                 skip_counter = 1
                 # TODO: use typing.get_type_hints when 3.9 is standard
                 typehints = typing_extensions.get_type_hints(self.callback, include_extras=True)
+                callback_params = signature(self.callback).parameters
+                possible_options: Set[str] = set()
+
+                for i, param in enumerate(callback_params.values()):
+                    if not (
+                        param.annotation is param.empty  # could be a self or interaction parameter
+                        or issubclass(
+                            param.annotation, Interaction
+                        )  # will always be an interaction parameter
+                        # TODO: use typing.Self when 3.11 is standard
+                        or param.annotation
+                        is typing_extensions.Self  # will always be a self parameter
+                        or isinstance(
+                            param.annotation, typing.TypeVar
+                        )  # will always be a self parameter
+                    ) and i in (0, 1):
+                        possible_options.add(param.name)
+
+                intersection = {param.name for param in callback_params.values()} - possible_options
+
+                if self.parent_cog is not None and len(intersection) < 2:
+                    raise ValueError(
+                        f"Callback {self.error_name} is missing the self and/or interaction parameters. Please double check your function definition."
+                    )
+                elif len(intersection) < 1:
+                    raise ValueError(
+                        f"Callback {self.error_name} is missing the interaction parameter. Please double check your function definition."
+                    )
+
+                # not needed later on
+                del possible_options, intersection
```

Comment:
I don't really see the point of deleting variables here, I think Python handles freeing local variables automatically when the function finishes and these aren't taking up a lot of memory.